### PR TITLE
Issue #140 GET /v4/projects endpoint always returns the attachments data

### PR DESCRIPTION
**Pull request by narekcat**
#140 GET /v4/projects endpoint always returns the attachments data

### CODE REVIEW

File: src/routes/projects/list.js [hunk 1]
```diff
@@ -134,7 +134,10 @@ const parseElasticSearchCriteria = (criteria, fields, order) => {
     const phaseFields = _.get(fields, 'project_phases_products');
     sourceInclude = sourceInclude.concat(_.map(phaseFields, single => `phases.products.${single}`));
   }
-  sourceInclude = sourceInclude.concat(_.map(PROJECT_ATTACHMENT_ATTRIBUTES, single => `attachments.${single}`));
+  if (_.get(fields, 'attachments', null)) {
+    const phaseFields = _.get(fields, 'attachments');
```
Comment (vikasrohit) [inline src/routes/projects/list.js]: I think copy paste magic is here. We need to rename the `phaseFields`.

File: src/routes/projects/list.spec.js [hunk 2]
```diff
@@ -306,6 +306,83 @@ describe('LIST Project', () => {
...
+          .expect('Content-Type', /json/)
+          .expect(200)
+          .end((err, res) => {
+            if (err) {
+              done(err);
+            } else {
+              const resJson = res.body.result.content;
+              should.exist(resJson);
+              resJson.should.have.lengthOf(3);
+              resJson[0].should.have.property('attachments');
```
Comment (vikasrohit) [inline src/routes/projects/list.spec.js]: It would be great if you can validate the fields of attachment objects as well, in one unit test.

**Comment by vikasrohit** [comment]
@narekcat I am not sure if our challenge guideline mentioned this or not, but for future bug bash challenges, please create a fork of the repo and create PR from your fork to the origin.

File: src/routes/projects/list.js (hunk 1 above)
Comment (narekcat) [reply src/routes/projects/list.js]: Yes, it is really copy paste magic.

**Comment by narekcat** [comment]
@vikasrohit In the challenge guideline didn't mention, but for future challenges I'll create fork. And I'll try to squash changes in one commit for one issue.

File: src/routes/projects/list.spec.js (hunk 2 above)
Comment (narekcat) [reply src/routes/projects/list.spec.js]: Ok, I'll do.

Comment (narekcat) [reply src/routes/projects/list.spec.js]: @vikasrohit I have fixed this comment too.

**Comment by vikasrohit** [comment]
Thanks @narekcat. I would like to have the assertions for values of the fields instead of just checking existence of the properties. However, I think might go out of scope of original requirements, hence, accepting the PR. Will create a separate issue for adding field value validations.

@coderReview merging PR to unblock other PRs and resolve conflicts with other PRs. We can close the issue once we verify the changes in dev.

**Comment by narekcat** [comment]
@vikasrohit Sorry for that, but you wrote to add validations for attachments fields only. Ok, it isn't a problem, I'll add assertions for values of fields when you open issue.

**Comment by coderReview** [comment]
I will update wiki

**Pull request merged**